ecommerce/extensions/payment/processors/liqpay.py

```diff
@@ -0,0 +1,221 @@
+""" LiqPay payment processing. """
+from __future__ import unicode_literals
+
+import logging
+
+import hashlib
+import json
+import requests
+from base64 import b64decode, b64encode
+from decimal import Decimal
+from django.urls import reverse
+from oscar.apps.payment.exceptions import GatewayError
+from urlparse import urljoin
+
+from ecommerce.core.url_utils import get_ecommerce_url
+from ecommerce.extensions.checkout.utils import get_receipt_page_url
+from ecommerce.extensions.payment.exceptions import DuplicateReferenceNumber, InvalidSignatureError
+from ecommerce.extensions.payment.processors import BasePaymentProcessor, HandledProcessorResponse
+
+
+logger = logging.getLogger(__name__)
+
+
+class Liqpay(BasePaymentProcessor):
+    """
+    Constructs a new instance of the Liqpay processor (October 2019)
+
+    For reference, see https://www.liqpay.ua/documentation/en/api/home/.
+    """
+
+    NAME = 'liqpay'
+
+    def __init__(self, site):
+        """
+        Constructs a new instance of the Liqpay processor.
```

Comment:
Just for your information, that `__init__` is not a constructor in python's world.